### PR TITLE
Don't validate constants in const propagation

### PR DESCRIPTION
**Pull request by tmiasko**
Validation is neither necessary nor desirable.

The constant validation is already omitted at mir-opt-level >= 3, so there there are not changes in MIR test output (the propagation of invalid constants is covered by an existing test in tests/mir-opt/const_prop/invalid_constant.rs).

### CODE REVIEW

File: compiler/rustc_const_eval/src/interpret/validity.rs [hunk 1]
```diff
@@ -36,16 +36,14 @@ macro_rules! throw_validation_failure {
             msg.push_str(", but expected ");
             write!(&mut msg, $($expected_fmt)*).unwrap();
         )?
-        let path = rustc_middle::ty::print::with_no_trimmed_paths!({
```
Comment (RalfJung) [inline compiler/rustc_const_eval/src/interpret/validity.rs:39]: Oh, I hadn't even realized that this was added in the validity checker. Happy to see it gone!

**Comment by RalfJung** [comment]
I'm happy with the interpreter value validity checker change, but can't really comment on the ConstProp diff -- I have no idea why this check was added, and whether it was ever needed for correctness.

**Comment by saethlin** [comment]
I do not understand the justification for this change. It doesn't make sense to me to justify changing the default behavior of the compiler based on the fact that we test in a special configuration with different behavior.

**Comment by apiraino** [comment]
hello visiting this PR, checking progress.

@tmiasko can you share some thoughts about [this comment ](https://github.com/rust-lang/rust/pull/109521#issuecomment-1481183071)? thanks

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
I looked through the history and couldn't figure out why I added the validation either. I also checked Zulip to see if there was some discussion or PMs around that time but didn't find anything. 

@bors r+

**Pull request merged**